src/core/session.cpp

```diff
@@ -26,6 +26,15 @@ void ensureConnected(session_backend * backEnd)
     }
 }
 
+struct query_parameter
+{
+    query_parameter(std::string name = {}, std::string value = {})
+        : name(std::move(name)), value(std::move(value)) {}
+
+    std::string name;
+    std::string value;
```

Comment:
Not that important, but I'd make this change

```suggestion
    const std::string name;
    const std::string value;
```

to be explicit about the fact that they never change.

Reply:
I think here is the wrong place to do this. This should be encoded in the place that actually uses an instance of this struct by making that specific instance `const`.

Additionally, having const members can have unintended side-effects like a disabled (by default) assignment operator. That's why I usually don't like having `const` members :thinking: